### PR TITLE
Data search

### PR DESCRIPTION
**Pull request by dcsaszar**
Content for equipment search: https://edit-beta.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://dave-data-search.scrivito-portal-app.pages.dev/en/equipment?_scrivito_workspace_id=s33bdf5568004455&_scrivito_display_mode=diff

The design of the input is just a copy of the header search. We could redesign the in-page search in a follow-up.

Heads up: The Pisa backend currently fails with a `400 Bad Request` b/c of the combination of `isCustomer` and fulltext `_search`. A patch is planned for 2024-07-02.

### CODE REVIEW

File: src/Components/DataBatchContext.tsx [hunk 1]
```diff
@@ -23,23 +27,27 @@ export const DataBatchContextProvider = connect(
     const configuredLimit = dataScope.limit() ?? 20
     const [limit, setLimit] = useState(configuredLimit)
     const [initialLimit, setInitialLimit] = useState(configuredLimit)
+    const [search, setSearch] = useState<string | undefined>()
```
Comment (apepper) [inline src/Components/DataBatchContext.tsx:30]: It looks like, that the `search` value sometimes is reused after navigating away from a page.

How to reproduce:
* Visit https://edit-beta.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://dave-data-search.scrivito-portal-app.pages.dev/en/equipment?_scrivito_workspace_id=s33bdf5568004455&_scrivito_display_mode=view
* Search for "clean"
* => only "Air cleaner" is shown
* Click on Orders

Expected: Open orders are visible.

Actual: No orders found.

File: src/Widgets/DataSearchWidget/DataSearchWidgetEditingConfig.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        { value: 'btn-primary', title: 'Primary color' },
+        { value: 'btn-secondary', title: 'Secondary color' },
+      ],
+    },
+    placeholder: { title: 'Placeholder text' },
+  },
+  properties: ['buttonColor', 'placeholder'],
+  initialContent: {
+    buttonColor: 'btn-primary',
+    buttonSize: 'medium',
```
Comment (apepper) [inline src/Widgets/DataSearchWidget/DataSearchWidgetEditingConfig.ts:22]: There is no `buttonSize` attribute for `DataSearchWidget`.

**Review by apepper — CHANGES_REQUESTED** [review]
FEEDBACK ([f15d02b](https://github.com/Scrivito/scrivito-portal-app/pull/458/commits/f15d02b02d7d3c209514602cf8efd2cfbb792630))

Nice start 🚀!

Content wise: Could you add search in more places? Gutfeeling: in tickets, quotes, orders and documents.

**Comment by dcsaszar** [comment]
REVIEW

> in tickets, quotes, orders and documents

All done but documents. When I tried to search I never got a result.

**Review by apepper — CHANGES_REQUESTED** [review]
FEEDBACK ([60dad5f](https://github.com/Scrivito/scrivito-portal-app/pull/458/commits/60dad5f8bf6c5f6b118f0f9c3a2dedef6539972e))

Nice 🚀!

There are still some details left. I also added search to documents and reported the 500 error to Ronald.

**Review by apepper — COMMENTED** [review]
FEEDBACK ([bfa07e2](https://github.com/Scrivito/scrivito-portal-app/pull/458/commits/bfa07e231d79b9401e45ed4fa8a3ac4c325dfd2e))

Code-wise really nice. I still need to clarify if Pisa-Sales supports all search cases.

**Pull request merged**